tests/test_transform.py

```diff
@@ -0,0 +1,231 @@
+import tempfile
+from pathlib import Path
+from typing import Optional
+
+import pytest
+import polars as pl
+
+from src.models.models import Config
+from src.transform import Transform
+from src.utils.utils import ETLUtils
+
+
+@pytest.fixture
+def temp_directories():
+    """
+    Pytest fixture to create temporary input and output directories.
+    """
+    with tempfile.TemporaryDirectory() as input_dir, tempfile.TemporaryDirectory() as output_dir:
+        yield Path(input_dir), Path(output_dir)
+
+
+@pytest.fixture
+def sample_csv_content():
+    """
+    Pytest fixture to provide sample CSV content with SEDOL and ISIN columns.
+    """
+    return r"""FINANCIAL TYPE,SYMBOL,SECURITY NAME,ISIN,PRICE,QUANTITY,REALISED P/L,MARKET VALUE
+            Equity,AAPL,Apple Inc.,US0378331005,150.00,10,500.00,1500.00
+            Equity,GOOGL,Alphabet Inc.,US02079K3059,2800.00,5,14000.00,14000.00
+            """.strip()
+
+
+@pytest.mark.parametrize(
+    "filename,expected_table_name,expected_date",
+    [
+        ("Applebead.30-06-2023 breakdown.csv", "applebead", "2023-06-30"),
+        ("Belaware.30_04_2023.csv", "belaware", "2023-04-30"),
+        ("Fund Whitestone.30-09-2022 - details.csv", "fund_whitestone", "2022-09-30"),
+        ("Leeder.04_30_2023.csv", "leeder", "2023-04-30"),
+        ("Magnum.30-09-2022.csv", "magnum", "2022-09-30"),
+        (
+            "mend-report Wallington.30_04_2023.csv",
+            "mend_report_wallington",
+            "2023-04-30",
+        ),
+        ("Report-of-Gohen.02-28-2023.csv", "report_of_gohen", "2023-02-28"),
+        ("rpt-Catalysm.2023-03-31.csv", "rpt_catalysm", "2023-03-31"),
+        ("TT_monthly_Trustmind.20220930.csv", "tt_monthly_trustmind", "2022-09-30"),
+        ("Virtous.05-31-2023 - securities.csv", "virtous", "2023-05-31"),
+        ("NoDateFile.csv", "no_date_file", None),
+        ("InvalidDate.99-99-9999.csv", "invalid_date", None),
+    ],
+)
+def test_extract_table_name_and_date(
+    temp_directories,
+    filename: str,
+    expected_table_name: Optional[str],
+    expected_date: Optional[str],
+):
+    """
+    Test the Transform.extract_table_name and ETLUtils.extract_date functions with various filename patterns.
+    Ensures table names are in snake_case and dates are correctly extracted.
+    """
+    input_dir, output_dir = temp_directories
+    config = Config(input_directory=input_dir, output_directory=output_dir)
+
+    # Extract table name using Transform class
+    extracted_table_name = ETLUtils.extract_table_name(filename)
+    assert (
```

Comment:
Good test, with a full set of possibilities in the parametrize. Note that test asserts don't normally have second arguments ("Failed table ...").